app/serializers/listable_topic_serializer.rb

```diff
@@ -9,6 +9,8 @@ class ListableTopicSerializer < BasicTopicSerializer
              :last_posted_at,
              :bumped,
              :bumped_at,
+             :archetype,
+             :is_pm,
```

Comment:
I don't think we need both of these new parameters. Can we manage with just `archetype`? That would match other places where topics are serialized. 

If you include the archetype, then you can [magically](https://github.com/discourse/discourse/blob/master/app/assets/javascripts/discourse/models/topic.js.es6#L352) use the `topic.isPrivateMessage` boolean in handlebars templates.

Reply:
I had no idea, thanks!  There seems to be a lot of black magic here that I don't yet understand :)